### ISSUE TITLE
RTD functions stops working

### ISSUE THREAD

**Issue by ShieldTrade** [cunnane/xloil#76]
Xloil version 0.17.14

Using the following code call `pyclock `then `xloil_gen `( order is important). Everything works as expected. You can change `xloil_gen `input and the function keeps working.

Now cancel `pyclock `and you will notice you can call `xloil_gen `only once. The second time you try to call it it returns #N/D. 
If you force a full recalculation you can see `xloil_gen ` is still alive ( alive might mean running on the xloil event loop but I am not skilled enough to assert it)  even if not updating. 
At this point, if you call `pyclock `or another RTD, they only update with a full recalculation.

Without `pyclock `, `xloil_gen `works only once.

At the moment my workaround is to wrap `df_result `in `generator(n)` in a `never ending while loop`.

```
while True:
        yield df_result
        await asyncio.sleep(2)
```

```
""" 
Generate n requests. 1 every 3 sec. 
Every request returns an incremental number. 
While waiting for the last request to be done show progress ( i.e which request has been sent). 
When all n numbers are received show the final result. 

The idea is to create a Minimal, Reproducible Example for simulating a long task (i.e fetching data from a server.) 
Since it might take a while we want to show progress and update them. Therefore we need to use RTD. 

Without the need to show progress we could have used a normal async function and return the result. 
"""

import asyncio, datetime
import xloil as xlo
import xloil.pandas
import pandas as pd

df_progress = pd.DataFrame()
df_waiting = pd.DataFrame([{'request': 'Waiting for data'}]).set_index('request')
df_result = None

@xlo.func( name='xloil_gen', group='UselessFuncs', help='returns its argument', args={'x': 'the argument'})
async def py_call_gen(n) -> pd.DataFrame:
    async for df in generator(n): yield (df)

async def generator(n)-> pd.DataFrame:
    tasks = {}
    loop = xlo.get_event_loop()
    running_task = asyncio.run_coroutine_threadsafe(create_df(n), loop)
    # running_task = asyncio.create_task(create_df(n))  same behaviour
    tasks['create_df'] = running_task

    while not tasks['create_df'].done():
        if df_progress.empty: yield df_waiting
        else: yield df_progress
        await asyncio.sleep(0.5)

    yield df_result

async def create_df(n)-> pd.DataFrame:
    global df_progress
    global df_result
    d_result, request = { 'result': [], }, 1

    while request <= n:
        """  df_progress """
        d_running = {'request': request, 'progress': "Waiting for Data"}
        df_running = pd.DataFrame([d_running])
        df_progress = pd.concat([df_progress, df_running], ignore_index=True)
        """ df_result """
        d_result['result'].append("result " + str(request))
        request +=1
        await asyncio.sleep(3)

    df_result = pd.DataFrame(d_result)
    df_progress = pd.DataFrame() # reuse for next call

@xloil.func
async def pyClock():
    while True:
        yield datetime.datetime.now()
        await asyncio.sleep(5)
```

**Comment by ShieldTrade** [comment]
Hi @cunnane,

I experienced another similar problem which might help to debug this issue.

Just call the above` pyclock().` Then cause an error in the cell: like in` pyclock(a1)` or `'=pyclock()`. 
Now `pyclock()` or any other RTD will not work anymore. But if you force a full recalculate you can see it is still alive.

if 2 cells call `pyclock()` and only 1 gets the error the problem disappear.

**Comment by cunnane** [comment]
Thanks! This clearly illustrates the problem.  Fixed in 0.18.5
